nucypher/policy/conditions/lingo.py

```diff
@@ -104,21 +112,48 @@ def __init__(self, comparator: str, value: Any):
 
         self.comparator = comparator
         self.value = value
+        self.key = key
 
     def _sanitize_value(self, value):
         try:
             return ast.literal_eval(str(value))
         except Exception:
             raise self.InvalidExpression(f'"{value}" is not a permitted value.')
 
+    def _process_data(self, data: Any) -> Any:
+        """
+        If a key is specified, return the value at that key in the data if data is a dict or list-like.
+        Otherwise, return the data.
+        """
+        if self.key and isinstance(data, dict):
+            try:
+                processed_data = data[self.key]
+            except KeyError:
+                raise ReturnValueEvaluationError(
+                    f"Key '{self.key}' not found in return data."
+                )
+        elif isinstance(self.key, int) and isinstance(data, (list, tuple)):
+            try:
+                processed_data = data[self.key]
+            except IndexError:
+                raise ReturnValueEvaluationError(
+                    f"Index '{self.key}' not found in return data."
+                )
+        else:
+            processed_data = data
```

Comment:
I think we chatted about this previously - but I don't remember the conclusion.

If `key` is specified, and the data instance is not tuple, list, or dict then then a `ReturnValueEvaluationError` should be raised, and not just continue on with the `data` as is - this case is not currently handled.

If `key` is not specified then obviously we do `processed_data = data`, and should do this. See prior comment here - https://github.com/nucypher/nucypher/pull/3002#discussion_r1019268711

Reply:
this is really strange, I definitely made this change. I think something may have gotten overwritten when I rebased

Reply:
Yeah I thought I remembered you did...damn rebases 😬

Reply:
I very confused, I have 100% already fixed this :D i think i may have overwritten a bunch of stuff when rebasing

eugh

Reply:
and now my comments aren't adding correctly...